File: Application/Classes/Controllers/TabProfiles/PortFolio/PortFolioCells/CompletedCell.m
//
//  CompletedCell.m
//  CoAssets
//
//  Created by Macintosh HD on 1/15/16.
//  Copyright © 2016 Sanyi. All rights reserved.
//

#import "CompletedCell.h"
#import "COCompleteModel.h"

@interface CompletedCell()
{
    __weak IBOutlet UILabel *_lblcurrency;
    __weak IBOutlet UILabel *_lblTextfooter;
}

@end
@implementation CompletedCell

- (void)awakeFromNib {
    [super awakeFromNib];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
    _lblTextfooter.text = m_string(@"MESSAGE_COMPLETE_PORTPOLIO");
}

#pragma mark - Set Get
- (void)setArrayComplete:(NSArray *)arrayComplete {
    _arrayComplete = arrayComplete;
    NSString *_strText;
    for (COCompleteModel *model in arrayComplete) {
        NSString *key = model.keyComplete;
        NSString *value = [UIHelper formartFoatValueWithPortfolio: model.valueComplete];
        NSString *content = [key stringByAppendingFormat:@"%@%@",@"-",value];
        if (_strText != nil) {
            _strText = [_strText stringByAppendingFormat:@"\n%@",content];
        } else {
            _strText = content;
        }
    }
    _lblcurrency.text = _strText;
}

@end

File: CoAssetsApps/Targets/Investor/InvestorCore/Classes/Controllers/ProfileFlow/Portfolio/PortfolioCells/CompletedCell.swift
//
//  CompletedCell.swift
//  CoAssetsApps
//
//  Created by Nikmesoft_M008 on 03/03/2016.
//  Copyright © 2016 TruongVO07. All rights reserved.
//

import UIKit

class CompletedCell: UITableViewCell {

    @IBOutlet weak private var currencyLabel: BaseLabel!
    @IBOutlet weak private var textFooterLabel: BaseLabel!
    
    var completeArray: [COPortfolioCompletedModel]? {
        didSet {
            refreshUI()
        }
    }
    override func awakeFromNib() {
        super.awakeFromNib()
        textFooterLabel.text = m_local_string("portfolio_Completed")
    }
    func refreshUI() {
        if let completeArray = completeArray {
            var text: String?
            for completeItem in completeArray {
                let key = completeItem.keyComplete
                let value  = FormattedHelper.formartFoatValueWithPortfolio(completeItem.valueComplete)
                let contentString = key.stringByAppendingFormat("%@%@", "-", value)
                if let _ = text {
                    text = text!.stringByAppendingFormat("\n%@", contentString)
                } else {
                    text = contentString
                }
            }
            kMainQueue.addOperationWithBlock({ () -> Void in
                self.currencyLabel.text = text
            })
        }
    }
}
